allergies/allergies.js

```diff
@@ -0,0 +1,66 @@
+//empty arrays for the user inputs to go into
+var healthRestrictions = [];
+var dietRestrictions = [];
+
+/*click event for all elements with the "healthIcon" class, if clicked it updates the class to 
+ reflect that, changes the image to a purple version as needed, and updates the healthRestrictions arrays*/ 
+$('.healthIcon').on('click', function() {
+    if ($(this).hasClass('notClicked')) {
+
+        healthRestrictions.push($(this).attr('value'));
+
+        $('#' + $(this).attr('value') + 'Icon').attr('src', '../assets/purp icons/' + $(this).attr('value') + '.png');
+        $(this).addClass('clicked');
+        $(this).removeClass('notClicked');
+    }
+    else if ($(this).hasClass('clicked')) {
+
+        var index1 = healthRestrictions.indexOf($(this).attr('value'));
+        if (index1 > -1) {
+            healthRestrictions.splice(index1, 1);
+        }
+
+        $('#' + $(this).attr('value') + 'Icon').attr('src', '../assets/white icons/' + $(this).attr('value') + '.png');
+        $(this).addClass('notClicked');
+        $(this).removeClass('clicked');
+    }
+});
+
+/*click event for all elements with the "dietIcon" class, if clicked it updates the class to 
+ reflect that, changes the image to a purple version as needed, and updates the dietRestrictions arrays*/ 
+$('.dietIcon').on('click', function() {
+    if ($(this).hasClass('notClicked')) {
+
+        dietRestrictions.push($(this).attr('value'));
+
+        $('#' + $(this).attr('value') + 'Icon').attr('src', '../assets/purp icons/' + $(this).attr('value') + '.png');
+        $(this).addClass('clicked');
+        $(this).removeClass('notClicked');
+    }
+    else if ($(this).hasClass('clicked')) {
```

Comment:
this can probably just be an else... your only options are clicked and not clicked, right?

you want one of these two things to happen no matter what so just having an else gives you a little protection and insurance that you're not somehow going to fall through to a case where no code is executed